src/Tracing/DynamicSamplingContext.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tracing;
+
+use Sentry\State\HubInterface;
+use Sentry\State\Scope;
+
+/**
+ * This class represents the Dynamic Sampling Context (dsc).
+ *
+ * @see https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/
+ */
+final class DynamicSamplingContext
+{
+    public const SENTRY_ENTRY_PREFIX = 'sentry-';
+
+    /**
+     * The dsc entries.
+     *
+     * @var array<string, string>
+     */
+    private $entries;
+
+    /**
+     * Indicates that the dsc is frozen and cannot be mutated.
+     *
+     * @var bool
+     */
+    private $frozen = false;
+
+    /**
+     * Construct a new dsc object.
+     */
+    private function __construct()
+    {
+        $this->entries = [];
+    }
+
+    /**
+     * Set a new key value pair on the dsc.
+     *
+     * @param string $key   the list member key
+     * @param string $value the list member value
+     */
+    public function set(string $key, string $value): void
+    {
+        if ($this->frozen) {
+            return;
```

Comment:
nit: I don't know what would be the use case where people try to set this when frozen but I think it may be valuable to print something in the logs so users are aware why the context is not set.

Reply:
Once the DSC is populated for the first time, we must make it immutable. This implements this.
As people should not mess around with the DSC from the outside, this API won't be documented in the docs as well.

Reply:
I wonder if we should throw an exception rather than silently doing nothing... I mean, if changing the context after it has been frozen is an error then it should be reported as such

Reply:
We should be a bit careful when to throw/trigger errors.